### PR TITLE
remove some redundant async state machines

### PR DESCRIPTION
**Pull request by SimonCropp**
#skip-changelog

### CODE REVIEW

File: src/Sentry/Internal/BackgroundWorker.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ internal class BackgroundWorker : IBackgroundWorker, IDisposable
             _shutdownSource = shutdownSource ?? new CancellationTokenSource();
             _queuedEnvelopeSemaphore = new SemaphoreSlim(0, _maxItems);
 
-            WorkerTask = Task.Run(async () => await WorkerAsync().ConfigureAwait(false));
+            WorkerTask = Task.Run(() => WorkerAsync());
```
Comment (josh-degraw) [inline src/Sentry/Internal/BackgroundWorker.cs:49]:
```suggestion
            WorkerTask = Task.Run(WorkerAsync);
```
Might as well simplify this further

**Comment by lucas-zimerman** [comment]
Could you add `#skip-changelog` to the PR description?

File: src/Sentry.AspNetCore/SentryMiddleware.cs [hunk 2]
```diff
@@ -88,10 +88,10 @@ public async Task InvokeAsync(HttpContext context)
 
                 if (_options.FlushOnCompletedRequest)
                 {
-                    context.Response.OnCompleted(async () =>
+                    context.Response.OnCompleted(() =>
```
Comment (bruno-garcia) [inline src/Sentry.AspNetCore/SentryMiddleware.cs:91]: Could be then:

```
// Serverless environments flush the queue at the end of each request
context.Response.OnCompleted(() => hub.FlushAsync(timeout: _options.FlushTimeout));
```

**Review by bruno-garcia — COMMENTED** [review]
One downside of returning tasks instead of awaiting is the resulting stack trace in case of an error. It's easier to debug when we await on tasks since we can tell exactly where it failed.

**Comment by bruno-garcia** [comment]
I would use the `return Task` approach only on tests and sample code but not sure it's ideal on code we'll need to debug through from customers reported stack traces

**Comment by SimonCropp** [comment]
> not sure it's ideal on code we'll need to debug through from customers reported stack traces

I heave hear this fear from a few people. TBH i have never had any issue with returning a task in the 100+ projects i manage. And on a side not, all the MS libraries directly return a task where possible.

But i understand the reticence, so i will close this one

**Comment by bruno-garcia** [comment]
https://twitter.com/haacked/status/1441454691087310859?s=20

**Pull request closed without merging**